src/main/java/bio/terra/cli/command/resource/update/GcsObject.java

```diff
@@ -46,10 +57,28 @@ protected void execute() {
     // all update parameters are optional, but make sure at least one is specified
     if (!resourceUpdateOptions.isDefined()
         && newObjectName == null
-        && newBucketName.getNewBucketName() == null) {
+        && newBucketName == null
+        && newGcsPath == null) {
       throw new UserActionableException("Specify at least one property to update.");
     }
 
+    // parsing the path as bucket name and object name
+    if (newGcsPath != null) {
+      if (newBucketName != null || newObjectName != null) {
+        throw new UserActionableException(
+            "No need to specify --new-bucket-name and/or --new-object-name when --new-path is specified.");
```

Comment:
I found this case is different from add-ref. 
In add-ref, the `--bucket-name` should always with `--object-name`. but in update, it can separate use.
But could not use `--new-path` while use any one of --new-bucket-name and --new-object-name. 
So I modify the exception throw to make it accurate. and add one more test for it.